src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -138,12 +138,12 @@ protected virtual Task<Document> GetChangedDocumentAsync(CancellationToken cance
         /// <summary>
         /// used by batch fixer engine to get new solution
         /// </summary>
-        internal async Task<Solution> GetChangedSolutionInternalAsync(CancellationToken cancellationToken)
+        internal async Task<Solution> GetChangedSolutionInternalAsync(bool postProcessChanges = true, CancellationToken cancellationToken = default(CancellationToken))
         {
             var solution = await GetChangedSolutionAsync(cancellationToken).ConfigureAwait(false);
-            if (solution == null)
+            if (solution == null || !postProcessChanges)
             {
-                return null;
+                return solution;
             }
 
             return await this.PostProcessChangesAsync(solution, cancellationToken).ConfigureAwait(false);
```

Comment:
Yes, but the code action itself might override that method. This API with flag basically guarantees that the caller will get the changed solution without running Formatter or Simplifier.